Public/Src/Utilities/Utilities/CommandLineEscaping.cs

```diff
@@ -235,16 +235,29 @@ private static int Repeat(StringBuilder builder, string s, int count)
 
         private static bool ContainsDelimiter(string s)
         {
-            foreach (char c in s)
+            if (OperatingSystemHelper.IsUnixOS)
             {
-                if (c <= ' ')
+                foreach (char c in s)
                 {
-                    if (c == ' ' || c == '\t')
+                    if (c == ' ' || c == '\t' || c == ';')
                     {
                         return true;
                     }
                 }
             }
+            else
+            {
+                foreach (char c in s)
+                {
+                    if (c <= ' ')

```

Comment:
It's hard for me to believe that this optimization has any noticeable effect, but since it was already there, I didn't want to get rid of it when running on Windows (hence the copy-pasted `foreach` loops)